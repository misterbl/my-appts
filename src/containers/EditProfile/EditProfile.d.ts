import { FormikProps } from 'formik';
import { RouteComponentProps, StaticContext } from 'react-router';
import { InjectedIntlProps } from 'react-intl';
import { ActionCreatorsMapObject } from 'redux';



export interface EditProfileDispatchToProps {
  apiThunk: ActionCreatorsMapObject;
}

export interface EditProfileState {
  selected: string
}
// export interface EditProfileMapStateToProps {
//   user: IAppState['api']['userData'];
// }

export type EditProfileComponent = InjectedIntlProps &
  RouteComponentProps<any, StaticContext>;
