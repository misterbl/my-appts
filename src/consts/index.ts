import ROUTES from './routes';
import QUERIES from './queries';

export { ROUTES, QUERIES }