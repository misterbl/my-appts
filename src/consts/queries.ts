const queries = (args: any) => ({
  ADD_USER: `mutation {
    addUser (email: "${args.email}", avatar:"${args.avatar}" ) {
      _id
      firstName
      lastName
      avatar
      email
      address
      profileTitle
      profileDescription
     
    }
  }`,
  GET_USER: `mutation {
    getUser (email: "${args.email}") {
      _id
      firstName
      lastName
      avatar
      email
      address
      lat
      lng
      drivingLicense
      nonSmoker
      car
      profileTitle
      profileDescription
      children{id, name, dob, information, school, gender}
      availability{beforeSchool, morning, noon, afternoon, afterSchool, evening, night}
      unavailability
    }
  }`,
  GET_USER_BY_ID: `mutation {
    getUserById (_id: "${args.id}") {
      _id
      firstName
      lastName
      avatar
      email
      address
      lat
      lng
      drivingLicense
      nonSmoker
      car
      profileTitle
      profileDescription
      children{id, name, dob, information, school, gender}
      availability{beforeSchool, morning, noon, afternoon, afterSchool, evening, night}
    }
  }`,
  GET_USER_INFO: `mutation {
    getUser (email: "${args.email}") {
      _id
      firstName
      lastName
      avatar
      email
      address
    }
  }`,
  GET_AD_INFO: `mutation {
    getUser (email: "${args.email}") {
      _id
      profileTitle
      profileDescription
    }
  }`,
  UPDATE_AD_INFO: `mutation {
    updateUser (_id:"${args._id}" profileTitle: "${
    args.profileTitle
  }", profileDescription: "${args.profileDescription}") {
      _id
      firstName
      lastName
      avatar
      address
      profileTitle
      profileDescription
      children{name, dob, information, school, gender} 
      unavailability
      availability{beforeSchool, morning, noon, afternoon, afterSchool, evening, night}
    }
  }`,
  UPDATE_PERSONAL_INFO: `mutation {
    updateUser (_id: "${args._id}", firstName: "${
    args.firstName
  }", lastName: "${args.lastName}", address: "${args.address}" ) {
      _id
      firstName
      lastName
      avatar
      address
      lat
      lng
      drivingLicense
      nonSmoker
      car
      profileTitle
      profileDescription
      children{name, dob, information, school, gender}
      unavailability
      availability{beforeSchool, morning, noon, afternoon, afterSchool, evening, night}

    }
  }`,
  UPDATE_AVATAR: `mutation {
    updateUser (_id: "${args._id}", avatar:"${args.avatar}" ) {
      _id
      firstName
      lastName
      avatar
      address
      lat
      lng
      drivingLicense
      nonSmoker
      car
      profileTitle
      profileDescription
      children{name, dob, information, school, gender}
      availability{beforeSchool, morning, noon, afternoon, afterSchool, evening, night}

    }
  }`,
  UPDATE_LOCATION: `mutation {
    updateUser (_id: "${args._id}", address: "${args.address}", lat: "${
    args.lat
  }", lng: "${args.lng}" ) {
      _id
      firstName
      lastName
      avatar
      address
      lat
      lng
      drivingLicense
      nonSmoker
      car
      profileTitle
      profileDescription
      children{name, dob, information, school, gender}
      unavailability
      availability{beforeSchool, morning, noon, afternoon, afterSchool, evening, night}

    }
  }`,

  UPDATE_ABOUT_YOU: `mutation {
    updateUser (_id: "${args._id}", drivingLicense: ${
    args.drivingLicense
  }, nonSmoker: ${args.nonSmoker}, car:${args.car}) {
      _id
      firstName
      lastName
      avatar
      address
      drivingLicense
      nonSmoker
      car
      profileTitle
      profileDescription
      children{name, dob, information, school, gender}
      availability{beforeSchool, morning, noon, afternoon, afterSchool, evening, night}
      unavailability

    }
  }`,
  UPDATE_AVAILABILITY: `mutation {
    updateUser (_id: "${args._id}", availability: ${JSON.stringify(
    JSON.stringify(args.availability),
  )}) {
      _id
      firstName
      lastName
      avatar
      address
      drivingLicense
      nonSmoker
      car
      profileTitle
      profileDescription
      unavailability
      availability{beforeSchool, morning, noon, afternoon, afterSchool, evening, night}
    }
  }`,
  UPDATE_UNAVAILABILITY: `mutation {
    updateUser (_id: "${args._id}", unavailability: ${JSON.stringify(
    args.unavailability,
  )}) {
      _id
      firstName
      lastName
      avatar
      address
      drivingLicense
      nonSmoker
      car
      profileTitle
      profileDescription
      availability{beforeSchool, morning, noon, afternoon, afterSchool, evening, night}
      unavailability
    }
  }`,
  ADD_CHILD: `mutation {
    updateUser (_id: "${args._id}", child: ${JSON.stringify(
    JSON.stringify(args.child),
  )}
) {
      _id
      firstName
      lastName
      avatar
      address
      profileTitle
      profileDescription
      children{id, name, dob, information, school, gender}
    }
  }`,
  UPDATE_CHILD: `mutation {
    updateChild (_id: "${args._id}", child: ${JSON.stringify(
    JSON.stringify(args.child),
  )}
) {
      _id
      firstName
      lastName
      avatar
      address
      profileTitle
      profileDescription
      children{id, name, dob, information, school, gender}
      availability{beforeSchool, morning, noon, afternoon, afterSchool, evening, night}
      unavailability
    }
  }`,
  REMOVE_CHILD: `mutation {
    removeChild (_id: "${args._id}", child: ${JSON.stringify(
    JSON.stringify(args.child),
  )}
) {
      _id
      firstName
      lastName
      avatar
      address
      profileTitle
      profileDescription
      children{id, name, dob, information, school, gender}
    }
  }`,
});

export default queries;
