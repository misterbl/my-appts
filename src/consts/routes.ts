const routes = {
  INDEX: '/',
  CARD: '/card',
  SIGN_IN: '/signin',
  REGISTER: '/register',
  DASHBOARD: './dashboard',
  INBOX: './inbox',
  FAVOURITE: './favourite',
  SEARCH: './search',
};

export default routes;
