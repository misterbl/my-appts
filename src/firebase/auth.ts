import { auth, provider } from './firebase';

// Sign Up
export const doCreateUserWithEmailAndPassword = (email: string, password: string) =>
  auth.createUserWithEmailAndPassword(email, password);

  // Sign In
export const doSignInWithEmailAndPassword = (email: string, password: string) =>
auth.signInWithEmailAndPassword(email, password);

// Sign out
export const doSignOut = () =>
  auth.signOut();


  // Password Reset
export const doPasswordReset = (email: string) =>
auth.sendPasswordResetEmail(email);

// Password Change
export const doPasswordUpdate = (password: string) =>
auth && auth.currentUser && auth.currentUser.updatePassword(password);

// Facebook
export const doFacebookSignIn = () => auth.signInWithPopup(provider).then(function(result) {
    // This gives you a Facebook Access Token. You can use it to access the Facebook API.
    const token = result;
    // The signed-in user info.
    const user = result.user;
    console.log("user", user);
    console.log("token", token);
    
    
    // ...
  }).catch(function(error) {
    // Handle Errors here.
    const errorCode = error.code;
    const errorMessage = error.message;
    console.log("errorCode", errorCode);
    console.log("errorMessage", errorMessage);
    
    // The email of the user's account used.
    const email = error.email;
    console.log("email", email);
    
    // The firebase.auth.AuthCredential type that was used.
    const credential = error.credential;
    console.log("credential", credential);
    
    // ...
  });