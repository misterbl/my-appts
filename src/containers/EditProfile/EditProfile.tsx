import * as React from 'react';
import { injectIntl, FormattedMessage } from 'react-intl';
import { withRouter } from 'react-router-dom';
import {
  TEditProfile,
  IEditProfileState,
  IEditProfileMapStateToProps,
} from './EditProfile.d';
import { ROUTES } from '../../consts';
import { chevronLeftIcon } from 'src/styles/assets';
import Svg from 'src/components/Svg';
import PersonalInfoForm from '../PersonalInfoForm/PersonalInfoForm';
import AddInfoForm from '../AddInfoForm/AddInfoForm';
import ChildrenForm from 'src/components/ChildrenForm/ChildrenForm';
import AvailabilityForm from 'src/components/AvailabilityForm/AvailabilityForm';
import { IAppState } from 'src/types/state';
import { getUserData } from 'src/selectors/apiSelectors';
import { connect } from 'react-redux';

export class EditProfile extends React.Component<
  TEditProfile,
  IEditProfileState
> {
  pagesName = ['myinfo', 'myad', 'mychildren', 'myavailability'];
  buttonRef = React.createRef<HTMLButtonElement>();
  constructor(props: TEditProfile) {
    super(props);
    this.state = {
      selected: 'myinfo',
    };
  }

  submitForm = () => {
    const node = this.buttonRef.current;
    if (node) {
      node.click();
    }
  };

  submitButton = () => (
    <button ref={this.buttonRef} className="dn" type="submit" />
  );

  handleClick = async (tab: string) => {
    await this.submitForm();
    this.setState({ selected: tab });
  };

  renderPage = (selected: any) => {
    switch (selected) {
      case 'myinfo':
        // @ts-ignore
        return <PersonalInfoForm submitButton={this.submitButton()} />;
      case 'myad':
        // @ts-ignore
        return <AddInfoForm submitButton={this.submitButton()} />;
      case 'mychildren':
        // @ts-ignore
        return <ChildrenForm submitButton={this.submitButton()} />;
      case 'myavailability':
        // @ts-ignore
        return <AvailabilityForm submitButton={this.submitButton()} />;
      default:
        return <span />;
    }
  };

  render() {
    const {
      pagesName,
      state: { selected },
      props: {
        history: { push },
        intl: { formatMessage },
      },
      handleClick,
      renderPage,
    } = this;

    return (
      <>
        <header className="green-bg pt3 shadow-3">
          <div
            className="flex ma0 ph5-ns"
            onClick={() => {
              this.submitForm();
              push(`${this.props.user && this.props.user._id}/card`);
            }}
          >
            <Svg fill="white" Icon={chevronLeftIcon} />
            <span className="white ml3 ttc">
              <FormattedMessage id="content|editprofile|myprofile" />
            </span>
          </div>
          <div className="f6 mt3 flex flex-nowrap justify-between mh2 ph5-ns">
            {pagesName.map(tab => (
              <div
                key={tab}
                onClick={() => handleClick(tab)}
                className={`white pb2 f7 ttu ${
                  selected === tab ? 'bb b--orange' : ''
                }`}
              >
                {formatMessage({
                  id: `content|editprofile|${tab}`,
                })}
              </div>
            ))}
          </div>
        </header>
        {renderPage(selected)}
      </>
    );
  }
}

export const mapStateToProps = (
  state: IAppState,
): IEditProfileMapStateToProps => ({
  user: getUserData(state),
});
const injectIntlEditProfile = injectIntl(EditProfile);
export default withRouter(connect(mapStateToProps)(injectIntlEditProfile));
