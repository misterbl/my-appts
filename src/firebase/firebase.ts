import * as firebase from 'firebase/app';
import 'firebase/auth';


const config = {
    apiKey: "AIzaSyAUFtb8M2Pzfgpn48-F736NVgKlCcc2haQ",
    authDomain: "kidappi-app.firebaseapp.com",
    databaseURL: "https://kidappi-app.firebaseio.com",
    projectId: "kidappi-app",
    storageBucket: "kidappi-app.appspot.com",
    messagingSenderId: "600523020740"
  };

  if (!firebase.apps.length) {
    firebase.initializeApp(config);
  }

  const auth = firebase.auth();
  auth.useDeviceLanguage();
  const provider = new firebase.auth.FacebookAuthProvider()
export {
  auth,
  provider,
};