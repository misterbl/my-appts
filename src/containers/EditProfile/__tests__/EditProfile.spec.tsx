import * as React from 'react';
import { shallow } from 'enzyme';
import historyMock from '../../../testMocks/history.mock';
import { EditProfile } from '../EditProfile';
import IntlMock from '../../../testMocks/intl.mock';
import generateAppState from '../../../testMocks/appState.mock';
import { auth } from '../../../firebase';
import { ROUTES } from '../../../consts';

jest.mock('../../../firebase', () => ({
  auth: {
    doSignOut: jest.fn().mockReturnValue(Promise.resolve({})),
  },
}));

describe('EditProfile', () => {
  const props = {
    user: generateAppState().api.userData,
    userEmail: 'email',
    intl: IntlMock,
    history: historyMock,
    location: historyMock.location,
    match: {
      params: {},
      isExact: true,
      path: '',
      url: '',
    },
    submitForm: jest.fn(),
  };
  // @ts-ignore
  const wrapper = shallow(<EditProfile {...props} />);
  it('matches the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should call history.push on click', () => {
    const div = wrapper.find('div').at(0);
    div.simulate('click');
    expect(props.history.push).toHaveBeenCalledWith('id/card');
  // });
  // it('should call auth.doSignOut() on click', async () => {
  //   const div = wrapper.find('#signOut');
  //   await div.simulate('click');
  //   expect(auth.doSignOut).toHaveBeenCalled();
  // });
  // it('should call props.history.push on click', async () => {
  //   const div = wrapper.find('#signOut');
  //   await div.simulate('click');
  //   expect(props.history.push).toHaveBeenCalledWith(ROUTES.INDEX);
  // });
});
