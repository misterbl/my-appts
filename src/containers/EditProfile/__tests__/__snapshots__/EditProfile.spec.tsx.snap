// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditProfile matches the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <EditProfile
    history={
        Object {
            "action": "PUSH",
            "block": [Function],
            "createHref": [Function],
            "go": [Function],
            "goBack": [Function],
            "goForward": [Function],
            "length": 123,
            "listen": [Function],
            "location": Object {
              "hash": "",
              "key": "",
              "pathname": "",
              "search": "",
              "state": "",
            },
            "push": [Function],
            "replace": [Function],
          }
    }
    intl={
        Object {
            "defaultFormats": "",
            "defaultLocale": "",
            "formatDate": [Function],
            "formatHTMLMessage": [Function],
            "formatMessage": [Function],
            "formatNumber": [Function],
            "formatPlural": [Function],
            "formatRelative": [Function],
            "formatTime": [Function],
            "formats": "",
            "locale": "",
            "messages": Object {},
            "now": [Function],
          }
    }
    location={
        Object {
            "hash": "",
            "key": "",
            "pathname": "",
            "search": "",
            "state": "",
          }
    }
    match={
        Object {
            "isExact": true,
            "params": Object {},
            "path": "",
            "url": "",
          }
    }
    submitForm={[Function]}
    user={
        Object {
            "_id": "id",
            "address": "address",
            "availability": undefined,
            "avatar": "avatar",
            "car": false,
            "children": Array [
              Object {
                "dob": 122322,
                "gender": "boy",
                "id": "childId",
                "information": "childInformation",
                "name": "childName",
                "school": "school",
              },
            ],
            "city": "city",
            "drivingLicense": false,
            "email": "email",
            "firstName": "firstName",
            "lastName": "lastName",
            "lat": 1234,
            "lng": 123445,
            "nonSmoker": false,
            "photoURL": "photU\`rl",
            "postCode": "postCode",
            "profileDescription": "profileDescription",
            "profileTitle": "profileTitle",
            "unavailability": undefined,
          }
    }
    userEmail="email"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <header
          className="green-bg pt3 shadow-3"
>
          <div
                    className="flex ma0 ph5-ns"
                    onClick={[Function]}
          >
                    <Svg
                              Icon={[Function]}
                              fill="#FF6300"
                    />
                    <span
                              className="orange ml3 ttc"
                    >
                              <FormattedMessage
                                        id="content|editprofile|myprofile"
                                        values={Object {}}
                              />
                    </span>
          </div>
          <div
                    className="f6 mt3 flex flex-nowrap justify-between mh2 ph5-ns"
          >
                    <div
                              className="orange pb2 f7 ttu bb b--orange"
                              onClick={[Function]}
                    >
                              content|editprofile|myinfo
                    </div>
                    <div
                              className="orange pb2 f7 ttu "
                              onClick={[Function]}
                    >
                              content|editprofile|myad
                    </div>
                    <div
                              className="orange pb2 f7 ttu "
                              onClick={[Function]}
                    >
                              content|editprofile|mychildren
                    </div>
                    <div
                              className="orange pb2 f7 ttu "
                              onClick={[Function]}
                    >
                              content|editprofile|myavailability
                    </div>
          </div>
</header>,
        <Connect(InjectIntl(PersonalInfoForm))
          submitButton={
                    <button
                              className="dn"
                              type="submit"
                    />
          }
/>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="flex ma0 ph5-ns"
              onClick={[Function]}
>
              <Svg
                            Icon={[Function]}
                            fill="#FF6300"
              />
              <span
                            className="orange ml3 ttc"
              >
                            <FormattedMessage
                                          id="content|editprofile|myprofile"
                                          values={Object {}}
                            />
              </span>
</div>,
            <div
              className="f6 mt3 flex flex-nowrap justify-between mh2 ph5-ns"
>
              <div
                            className="orange pb2 f7 ttu bb b--orange"
                            onClick={[Function]}
              >
                            content|editprofile|myinfo
              </div>
              <div
                            className="orange pb2 f7 ttu "
                            onClick={[Function]}
              >
                            content|editprofile|myad
              </div>
              <div
                            className="orange pb2 f7 ttu "
                            onClick={[Function]}
              >
                            content|editprofile|mychildren
              </div>
              <div
                            className="orange pb2 f7 ttu "
                            onClick={[Function]}
              >
                            content|editprofile|myavailability
              </div>
</div>,
          ],
          "className": "green-bg pt3 shadow-3",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <Svg
                  Icon={[Function]}
                  fill="#FF6300"
/>,
                <span
                  className="orange ml3 ttc"
>
                  <FormattedMessage
                                    id="content|editprofile|myprofile"
                                    values={Object {}}
                  />
</span>,
              ],
              "className": "flex ma0 ph5-ns",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "Icon": [Function],
                  "fill": "#FF6300",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <FormattedMessage
                    id="content|editprofile|myprofile"
                    values={Object {}}
/>,
                  "className": "orange ml3 ttc",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "id": "content|editprofile|myprofile",
                    "values": Object {},
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": "span",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div
                  className="orange pb2 f7 ttu bb b--orange"
                  onClick={[Function]}
>
                  content|editprofile|myinfo
</div>,
                <div
                  className="orange pb2 f7 ttu "
                  onClick={[Function]}
>
                  content|editprofile|myad
</div>,
                <div
                  className="orange pb2 f7 ttu "
                  onClick={[Function]}
>
                  content|editprofile|mychildren
</div>,
                <div
                  className="orange pb2 f7 ttu "
                  onClick={[Function]}
>
                  content|editprofile|myavailability
</div>,
              ],
              "className": "f6 mt3 flex flex-nowrap justify-between mh2 ph5-ns",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": "myinfo",
                "nodeType": "host",
                "props": Object {
                  "children": "content|editprofile|myinfo",
                  "className": "orange pb2 f7 ttu bb b--orange",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": "content|editprofile|myinfo",
                "type": "div",
              },
              Object {
                "instance": null,
                "key": "myad",
                "nodeType": "host",
                "props": Object {
                  "children": "content|editprofile|myad",
                  "className": "orange pb2 f7 ttu ",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": "content|editprofile|myad",
                "type": "div",
              },
              Object {
                "instance": null,
                "key": "mychildren",
                "nodeType": "host",
                "props": Object {
                  "children": "content|editprofile|mychildren",
                  "className": "orange pb2 f7 ttu ",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": "content|editprofile|mychildren",
                "type": "div",
              },
              Object {
                "instance": null,
                "key": "myavailability",
                "nodeType": "host",
                "props": Object {
                  "children": "content|editprofile|myavailability",
                  "className": "orange pb2 f7 ttu ",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": "content|editprofile|myavailability",
                "type": "div",
              },
            ],
            "type": "div",
          },
        ],
        "type": "header",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "submitButton": <button
            className="dn"
            type="submit"
/>,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <header
            className="green-bg pt3 shadow-3"
>
            <div
                        className="flex ma0 ph5-ns"
                        onClick={[Function]}
            >
                        <Svg
                                    Icon={[Function]}
                                    fill="#FF6300"
                        />
                        <span
                                    className="orange ml3 ttc"
                        >
                                    <FormattedMessage
                                                id="content|editprofile|myprofile"
                                                values={Object {}}
                                    />
                        </span>
            </div>
            <div
                        className="f6 mt3 flex flex-nowrap justify-between mh2 ph5-ns"
            >
                        <div
                                    className="orange pb2 f7 ttu bb b--orange"
                                    onClick={[Function]}
                        >
                                    content|editprofile|myinfo
                        </div>
                        <div
                                    className="orange pb2 f7 ttu "
                                    onClick={[Function]}
                        >
                                    content|editprofile|myad
                        </div>
                        <div
                                    className="orange pb2 f7 ttu "
                                    onClick={[Function]}
                        >
                                    content|editprofile|mychildren
                        </div>
                        <div
                                    className="orange pb2 f7 ttu "
                                    onClick={[Function]}
                        >
                                    content|editprofile|myavailability
                        </div>
            </div>
</header>,
          <Connect(InjectIntl(PersonalInfoForm))
            submitButton={
                        <button
                                    className="dn"
                                    type="submit"
                        />
            }
/>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="flex ma0 ph5-ns"
                onClick={[Function]}
>
                <Svg
                                Icon={[Function]}
                                fill="#FF6300"
                />
                <span
                                className="orange ml3 ttc"
                >
                                <FormattedMessage
                                                id="content|editprofile|myprofile"
                                                values={Object {}}
                                />
                </span>
</div>,
              <div
                className="f6 mt3 flex flex-nowrap justify-between mh2 ph5-ns"
>
                <div
                                className="orange pb2 f7 ttu bb b--orange"
                                onClick={[Function]}
                >
                                content|editprofile|myinfo
                </div>
                <div
                                className="orange pb2 f7 ttu "
                                onClick={[Function]}
                >
                                content|editprofile|myad
                </div>
                <div
                                className="orange pb2 f7 ttu "
                                onClick={[Function]}
                >
                                content|editprofile|mychildren
                </div>
                <div
                                className="orange pb2 f7 ttu "
                                onClick={[Function]}
                >
                                content|editprofile|myavailability
                </div>
</div>,
            ],
            "className": "green-bg pt3 shadow-3",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <Svg
                    Icon={[Function]}
                    fill="#FF6300"
/>,
                  <span
                    className="orange ml3 ttc"
>
                    <FormattedMessage
                                        id="content|editprofile|myprofile"
                                        values={Object {}}
                    />
</span>,
                ],
                "className": "flex ma0 ph5-ns",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "Icon": [Function],
                    "fill": "#FF6300",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <FormattedMessage
                      id="content|editprofile|myprofile"
                      values={Object {}}
/>,
                    "className": "orange ml3 ttc",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "id": "content|editprofile|myprofile",
                      "values": Object {},
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  "type": "span",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <div
                    className="orange pb2 f7 ttu bb b--orange"
                    onClick={[Function]}
>
                    content|editprofile|myinfo
</div>,
                  <div
                    className="orange pb2 f7 ttu "
                    onClick={[Function]}
>
                    content|editprofile|myad
</div>,
                  <div
                    className="orange pb2 f7 ttu "
                    onClick={[Function]}
>
                    content|editprofile|mychildren
</div>,
                  <div
                    className="orange pb2 f7 ttu "
                    onClick={[Function]}
>
                    content|editprofile|myavailability
</div>,
                ],
                "className": "f6 mt3 flex flex-nowrap justify-between mh2 ph5-ns",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": "myinfo",
                  "nodeType": "host",
                  "props": Object {
                    "children": "content|editprofile|myinfo",
                    "className": "orange pb2 f7 ttu bb b--orange",
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": "content|editprofile|myinfo",
                  "type": "div",
                },
                Object {
                  "instance": null,
                  "key": "myad",
                  "nodeType": "host",
                  "props": Object {
                    "children": "content|editprofile|myad",
                    "className": "orange pb2 f7 ttu ",
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": "content|editprofile|myad",
                  "type": "div",
                },
                Object {
                  "instance": null,
                  "key": "mychildren",
                  "nodeType": "host",
                  "props": Object {
                    "children": "content|editprofile|mychildren",
                    "className": "orange pb2 f7 ttu ",
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": "content|editprofile|mychildren",
                  "type": "div",
                },
                Object {
                  "instance": null,
                  "key": "myavailability",
                  "nodeType": "host",
                  "props": Object {
                    "children": "content|editprofile|myavailability",
                    "className": "orange pb2 f7 ttu ",
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": "content|editprofile|myavailability",
                  "type": "div",
                },
              ],
              "type": "div",
            },
          ],
          "type": "header",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "submitButton": <button
              className="dn"
              type="submit"
/>,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
